### DISCUSSION TITLE
How to intercept before TLS?

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6833] [General]
Is there a way of intercepting the requests before mitmproxy starts TLS?
We want to filter the requests based on the URL and the payload size to see if we want to process it or not so that we minimize processing unnecessary data.
Currently, the exposed addon methods like request() or requestheaders() are all called after TLS termination.
But we want to do it before TLS to avoid the overhead.

**Comment by mhils** [comment]
> Is there a way of intercepting the requests before mitmproxy starts TLS?

No. TLS encrypts the entire request (URL, headers, ...). The only thing we may see before TLS is the hostname, which is either sourced from the TLS ClientHello _Server Name Indication Extension_ or the HTTP CONNECT call.
